ecs-cli/modules/cli/compose/project/project_parseV3_test.go

```diff
@@ -608,6 +619,157 @@ services:
 	assert.ElementsMatch(t, expectedEnv, web.Environment, "Expected Environment to match")
 }
 
+func TestParseV3HealthCheck(t *testing.T) {
+	// set up expected ContainerConfig values
+	wordpressCon := adapter.ContainerConfig{}
+	wordpressCon.Name = "wordpress"
+	wordpressCon.Command = []string{"echo \"hello world\""}
+	wordpressCon.Image = "wordpress"
+	wordpressCon.HealthCheck = &ecs.HealthCheck{
+		Command:  aws.StringSlice([]string{"CMD-SHELL", "curl -f http://localhost || exit 1"}),
+		Interval: aws.Int64(int64(90)),
+		Timeout:  aws.Int64(int64(10)),
+		Retries:  aws.Int64(int64(3)),
+	}
+
+	// set up file
+	composeFileString := `version: '3'
+services:
+  wordpress:
+    command:
+      - echo "hello world"
+    image: wordpress
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f http://localhost || exit 1"]
+      interval: 1m30s
+      timeout: 10s
+      retries: 3`
+
+	tmpfile, err := ioutil.TempFile("", "test")
+	assert.NoError(t, err, "Unexpected error in creating test file")
+
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write([]byte(composeFileString))
+	assert.NoError(t, err, "Unexpected error writing file")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Unexpected error closing file")
+
+	// add files to projects
+	project := setupTestProject(t)
+	project.ecsContext.ComposeFiles = append(project.ecsContext.ComposeFiles, tmpfile.Name())
+
+	// assert # and content of container configs matches expected
+	actualConfigs, err := project.parseV3()
+	assert.NoError(t, err, "Unexpected error parsing file")
+
+	assert.Equal(t, 1, len(*actualConfigs))
+
+	wp, err := getContainerConfigByName(wordpressCon.Name, actualConfigs)
+	assert.NoError(t, err, "Unexpected error retrieving wordpress config")
+	verifyContainerConfig(t, wordpressCon, *wp)
+}
+
+func TestParseV3HealthCheckAltSyntax(t *testing.T) {
+	// set up expected ContainerConfig values
+	wordpressCon := adapter.ContainerConfig{}
+	wordpressCon.Name = "wordpress"
+	wordpressCon.Command = []string{"echo \"hello world\""}
+	wordpressCon.Image = "wordpress"
+	wordpressCon.HealthCheck = &ecs.HealthCheck{
+		Command:  aws.StringSlice([]string{"CMD-SHELL", "curl -f http://localhost || exit 1"}),
+		Interval: aws.Int64(int64(90)),
+		Timeout:  aws.Int64(int64(10)),
+		Retries:  aws.Int64(int64(3)),
+	}
+
+	// set up file
+	composeFileString := `version: '3'
+services:
+  wordpress:
+    command:
+      - echo "hello world"
+    image: wordpress
+    healthcheck:
+      test: curl -f http://localhost || exit 1
```

Comment:
To avoid writing a separate test, you can add a healthcheck with different syntax to the 2nd (mysql) service of TestParseV3WithOneFile(); we already do this for Ports and Labels

Reply:
Sure!